Kadane_Algorithm/Kandane_Algorithm.dart

```diff
@@ -0,0 +1,57 @@
+/* Dart implementation of Kadane's Algorithm to find the Maximum Subarray Sum
+including the extra phase required when all the numbers in array are negative */
+
+import 'dart:io';
+
+int max(a, b) // Function that returns maximum between Two Numbers
+{
+    if(a > b) 
+        return a;
+    else 
+        return b;
+}
+
+int kadane( input , n ) // Function implementing Kadane's Algorithm (array contains at least one positive number)
+{
+    int currentmax = 0, maxsofar = 0;
+
+    for(int i = 0; i < n; i++)
+    {
+        currentmax = max( 0 , currentmax + input[i]);
+        maxsofar = max( maxsofar, currentmax );
+    }
+
+    return maxsofar; // Maximum subarray sum
+}
+
+void main() {
+    var maxsubarraysum;
+    var input = [-2, 1, -6, 4, -1, 2, 1, -5, 4]; // Input array
+    var n = input.length; // Size of array
+    var flag = 0; // Flag variable to check if all the numbers in array are negative or not
```

Comment:
Please add the comments before the line of code rather than the side. The length of characters in one line gets huge if you add comments on side and affects readability.